aproxy.go

```diff
@@ -0,0 +1,404 @@
+//go:build linux
+
+package main
+
+import (
+	"bufio"
+	"encoding/binary"
+	"errors"
+	"flag"
+	"fmt"
+	"golang.org/x/crypto/cryptobyte"
+	"io"
+	"log"
+	"log/slog"
+	"net"
+	"net/http"
+	"net/url"
+	"os"
+	"strings"
+	"sync"
+	"sync/atomic"
+	"syscall"
+	"unsafe"
+)
+
+var version = "0.2.0"
+
+type PrereadConn struct {
+	prereadStarted bool
+	prereadEnded   bool
+	prereadBuf     []byte
```

Comment:
Because this is `PrereadConn`, the `preread` prefix is probably unnecessary on these field names, and they could just be `started`, `ended`, `buf`.

Reply:
Renamed, thanks!